docs/custom_node_development.md

```diff
@@ -0,0 +1,125 @@
+CustomNodeTensor struct. Number of outputs is to be defined during the function execution in the `outputsCount` argument.
+
+Note that during the function execution all the output data buffers needs to be allocated. They will be released by OVMS after 
+the request processing is completed and returned to the user. The cleanup is triggered by calling `release` function 
+which also needs to be implemented in custom library.
+
+Execute function returns an integer which value defines the success (`0` value) or failure ( greater than 0). When the function 
+reports error, the pipeline execution is stopped and error is returned to the user. 
+
+### "getInputsInfo" function
+This function returns information about the metadata of the expected inputs. Returned CustomNodeTensorInfo object is used 
+to create a response for getModelMetadata calls. It is also used in the user request validation and the pipeline 
+configuration validation.
+
+Custom nodes can generate the results which have dynamic size depending on the input data and the custom node parameters.
+In such case function getInputsInfo should return value `0` on the dimension with dynamic size. It could be an input with
+variable resolution or a batch size. 
+
+### "getOutputInfo" function
+Similar to previous function but defining the outputs metadata.
+
+### "release" function
+This function is called by OVMS at the end of the pipeline processing. It clear all memory allocations used during the 
+node execution. This function should only call `free`. OVMS decides when to free and what to free.
+
+
+## Using OpenCV
+The custom node library can use any third-party dependencies which could be linked statically or dynamically.
+For simplicity there could be used OpenCV libraries which are already included in the OVMS docker image.
+Just add include statement like:
+```c++
+#include "opencv2/core.hpp"
+```
+
+## Building
+Custom node library can be compiled using any tool but for simplicity there are prepared a method based on a g++ compiled 
+inside a docker container. A simple example is described in this [Makefile](../src/custom_nodes/east_ocr/Makefile). 
+
+## Testing 
+Recommended method for tesing the custom library is via OVMS execution:
+- Compile the library using a docker container like configured in the Makefile. It will be exported to `lib` folder.
+- Prepare a pipeline configuration with the path custom node library compiled in the previous step
+- Start OVMS docker container
+- Submit a request to OVMS endpoint using a gRPC or REST client
+- Analyse the logs on the OVMS server
+
+For debugging steps refer the OVMS [developer guide](developer_guide.md)
+
+
+## Custom node examples 
+The best staring point for developing new custom nodes is via exploring or copying the examples.
+
+The fully functional custom nodes are in:
+- [east-resnet50 OCR custom node](../src/custom_nodes/east_ocr)
+
+Other examples are included in the unit tests:
```

Comment:
I would limit ourselves to mentioning the path to the custom nodes and 2/3 libraries that do something with positive results. I don;t think that mentioning mocks here is useful